packages/theme/src/cli/utilities/theme-fs.ts

```diff
@@ -46,28 +48,82 @@ const THEME_PARTITION_REGEX = {
 }
 
 export async function mountThemeFileSystem(root: string): Promise<ThemeFileSystem> {
-  const filesPaths = await glob(THEME_DIRECTORY_PATTERNS, {
+  const files = new Map<string, ThemeAsset>()
+  const eventEmitter = new EventEmitter()
+  function emitEvent<T extends ThemeFSEventName>(eventName: T, payload: ThemeFSEvent<T>['payload']) {
+    eventEmitter.emit(eventName, payload)
+  }
+
+  const initialFilesPromise = glob(THEME_DIRECTORY_PATTERNS, {
```

Comment:
Note this is now async so that we can do other things at the same time. Before checking the contents of files, we need to await `ready()` to ensure initial files are all included.